R/build_tree.R

```diff
@@ -1,4 +1,4 @@
-build_tree <- function(x, title, settings) {
+build_tree <- function(x, title = NULL, settings = NULL) {
```

Comment:
I thought it was:
* `hp_dataframe()` title/settings required
* `hp_spacyr()` title not required, settings optional (for styling)

What if we just add informative stops for `hp_dataframe()`?

Reply:
The thing is they are not actually optional by the time you get to `build_tree`. You just initialized the values in `hp_spacyr` before invoking `build_tree`.

Reply:
I invoked them in `build_tree()` because there should be no reason that the title changes right? So we should not even allow the user to pass a title. I agree about the settings part though, that needs to be changed and placed in the param, not in the function.

Reply:
If you try to run `df_dataframe` without a title things will break. And if you try to use `NULL` as title, the widget breaks.

In `hp_spacyr`, the title is not required from the user BUT it is automatically set in this chunk:
```
  x <- build_tree(
    x = transform_logical(spacy_df(.data)),
    title = .data,
    settings = spacy_default()
  )
```
It is still a mandatory param. And users may not want to use everything in `spacy_default()` either. Also, if they want to style the tree, the settings will have to be something like `settings = spacy_default() %>% add_styles()`, which should be a customizable param.

Reply:
Right, so I think `hp_dataframe()` just needs a `stop()` if the title/settings are NULL whereas `hp_spacyr()` only needs the settings/... params added. I think?!

Reply:
Yes that is correct. Also if anyone were to call `build_tree` directly, they should not be led to think that `title` and `settings` are optional.

Reply:
But build_tree isn't exported, we don't plan to encourage users to call this function right?

Reply:
Probably not. I just wanted to point out that title and settings are not optional for our own clarification's sake. No need to update!

Reply:
Perfect, kk thanks for clarifying!